lib/srv/ctx.go

```diff
@@ -617,6 +626,23 @@ func (c *ServerContext) Close() error {
 		return trace.Wrap(err)
 	}
 
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	// if there was a tty allocated, update the user accounting database
+	// with information that it's now closed
+	if c.termTTY != nil {
+		ttyName, err := utils.TtyName(c.termTTY)
```

Comment:
I'm not yet sure whether these can fail and why, but I was wondering whether it would not make sense to get these on init instead of in Close?

Reply:
Sure does. Will change that. The reason they can fail is if something kills the tty prematurely so it's not valid anymore.